models/timeswap/arbitrum/timeswap_arbitrum_lend.sql

```diff
@@ -0,0 +1,138 @@
+{{ config(
+    alias = 'lend',
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['transaction_hash', 'pool_pair', 'maturity', 'strike'],
+    post_hook='{{ expose_spells(\'["arbitrum"]\',
+                                "project",
+                                "timeswap",
+                                \'["raveena15, varunhawk19"]\') }}'
+    )
+}}
+
+SELECT
+  l.evt_tx_hash as transaction_hash,
+  l.evt_block_time as time,
+  l.isToken0 as token_0,
+  'Lend' as transaction_type,
+  l.maturity as maturity,
+  l.strike as strike, 
+  i.pool_pair as pool_pair,
+  i.chain as chain,
+  CAST(
+    CASE
+      WHEN CAST(l.isToken0 AS BOOLEAN) = true THEN CAST(l.tokenAmount AS DOUBLE) / power(10,i.token0_decimals)
+      ELSE CAST(l.tokenAmount AS DOUBLE) / power(10,i.token1_decimals)
+    END as DOUBLE
+  ) as token_amount,
+  CAST(
+    CASE
+      WHEN CAST(l.isToken0 AS BOOLEAN) = true THEN CAST(l.tokenAmount AS DOUBLE) / power(10,i.token0_decimals) * p.price
+      ELSE CAST(l.tokenAmount AS DOUBLE) / power(10,i.token1_decimals) * p.price
+    END as DOUBLE
+  ) as usd_amount
+  FROM {{ source('timeswap_arbitrum', 'TimeswapV2PeripheryUniswapV3LendGivenPrincipal_evt_LendGivenPrincipal') }} l
+  JOIN {{ ref('timeswap_arbitrum_pools') }} i ON CAST(l.maturity as VARCHAR(100)) = i.maturity and cast(l.strike as VARCHAR(100)) = i.strike
+  JOIN {{ source('prices', 'usd') }} p ON p.minute = date_trunc('minute', l.evt_block_time)
+  WHERE p.symbol=i.token0_symbol AND p.blockchain = 'arbitrum' AND l.isToken0 = true
```

Comment:
for attempts at improving runtime, can we move these conditions to the join conditions to prices, rather than where clause?

Reply:
Thanks for the suggestion. Updated